xcpretty init

nicknovitski opened the pull request:
Xcpretty is a command-line tool which fastlane invokes, but sometimes I use it directly.
~~I added the updateScript attribute to fastlane because it updates very frequently.~~


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- ~[ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

nicknovitski committed:
xcpretty: init at 0.3.0

nicknovitski commented:
Are updateScript attributes not the current best practice for updateable packages?  If not, I can remove them.

c0bw3b commented:
It should live under `pkgs/development/tools` instead of `pkgs/tools/admin`

Change the call from `all-packages.nix` accordingly.
And its position in the file: there is a `### DEVELOPMENT / TOOLS` section

nicknovitski commented:
Sure.  I was just following the positioning of fastlane, since I think of them as related tools.

nicknovitski commented:
Moved to `pkgs/development/tools/xcpretty` and the `DEVELOPMENT / TOOLS` section.

alyssais commented on pkgs/development/tools/xcpretty/default.nix:
```diff
@@ -0,0 +1,20 @@
+{ lib, bundlerEnv, ruby }:
+
+bundlerEnv rec {
+  inherit ruby;
```
I don't think this is necessary?

alyssais commented on pkgs/development/tools/xcpretty/update:
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p bash ruby bundler bundix
```
Ruby isn't necessary here.

alyssais commented on pkgs/development/tools/xcpretty/update:
```diff
@@ -0,0 +1,10 @@
+#! nix-shell -i bash -p bash ruby bundler bundix
+
+rm Gemfile.lock
+bundler install
```
If you do `bundle lock --update` here, you don't need to remove the Gemfile, and Bundler won't actually attempt to install the gems. This means that native dependencies won't cause problems, and you won't need the `~/.gem` hack, since all gem builds will happen under Nix through Bundix.

nicknovitski replied:
😮 cool

alyssais commented on pkgs/development/tools/xcpretty/update:
```diff
@@ -0,0 +1,10 @@
+
+if [ "clean" == "$1" ]; then
+  rm -r ~/.gem
+fi
```
You shouldn't need this if you're building with Bundix.

alyssais commented on pkgs/development/tools/xcpretty/default.nix:
```diff
@@ -0,0 +1,20 @@
+{ lib, bundlerEnv, ruby }:
+
+bundlerEnv rec {
```
This should probably be `bundlerApp`?. See [the manual](https://nixos.org/nixpkgs/manual/#sec-language-ruby) for the difference.

nicknovitski replied:
That makes sense.  I changed it to accept a `passthru` argument and to make the `exes` argument optional, defaulting to the `pname`.

alyssais commented on pkgs/development/tools/xcpretty/default.nix:
```diff
@@ -0,0 +1,20 @@
+    description     = "Flexible and fast xcodebuild formatter ";
+    homepage        = https://github.com/supermarin/xcpretty;
+    license         = licenses.mit;
+    maintainers     = with maintainers; [
```
This style is a little unusual, but ultimately your call I think.

nicknovitski replied:
Ah, you made me realize there was a trailing space in the description 😅

alyssais replied:
Heh, I didn't even notice that.

nicknovitski committed:
bundlerApp: take passthru

nicknovitski commented:
Updated to use `bundlerApp` (which I also changed a bit), changed the `updateScript` to recommendations and made it an inline string following a pattern in nix-pin, nix-update-source, and gup, which locks down the versions of bundler and bundix it uses.

alyssais commented on pkgs/development/ruby-modules/bundler-app/default.nix:
```diff
@@ -15,7 +15,7 @@
 , gemdir
   # Exes is the list of executables provided by the gems in the Gemfile
-, exes ? []
+, exes ? [ pname ]
```
Not sure how I feel about this… There's not necessarily any reason for the executable name to be the same as the package name. IMHO it should be explicit.

zimbatm replied:
As long as the build fails if it's not the case it should be alright. But yeah, one more heuristic.

nicknovitski replied:
No problem, I thought it was a safe default, but it's hardly necessary.

nicknovitski commented:
Reverted the addition of a default value for `bundlerApp`s `exes` argument.
